src/p_map.c

```diff
@@ -931,6 +931,7 @@ boolean P_TryMove(mobj_t *thing, fixed_t x, fixed_t y, boolean dropoff)
   // killough 11/98: simplified
 
   if (!(thing->flags & (MF_TELEPORT | MF_NOCLIP)))
+  {
     while (numspechit--)
```

Comment:
@fabiangreffrath I think it's the only place where `numspechit < 0`

Reply:
The actual crash happened here:

https://github.com/fabiangreffrath/woof/blob/1eaa8cdff02eacda59dd9cc94fc37c2c62aacaf5/src/p_enemy.c#L439-L441

In the case at hand `P_TryMove()` was called before `P_Move()`. The former "ate up" all the special lines and resulted in `numspechit` becoming `-1`. Thus it passed the `if (!numspechit)` check in the latter, was further reduced to `-2` and then crashed when used as index in the `spechit[]` array.

I'm not sure if the scenario could happen the other way round, though. That is, `P_Move()` is called before `P_TryMove()` and leaves `numspechit` at `-1`.

Reply:
Should we just add this?
```diff
diff --git a/src/p_enemy.c b/src/p_enemy.c
index fa82eafd..8dc7a68f 100644
--- a/src/p_enemy.c
+++ b/src/p_enemy.c
@@ -439,6 +439,7 @@ static boolean P_Move(mobj_t *actor, boolean dropoff) // killough 9/12/98
       for (good = false; numspechit--; )
         if (P_UseSpecialLine(actor, spechit[numspechit], 0, false))
 	  good |= spechit[numspechit] == blockline ? 1 : 2;
+      numspechit = 0;
 
       // [FG] compatibility maze here
       // Boom v2.01 and orig. Doom return "good"

```

Reply:
Yes, either this, or we turn `numspechit--` into `numspechit-- > 0` in both places.